src/shared/lib/timers/model/TimerBase.ts

```diff
@@ -0,0 +1,67 @@
+import { AppState, NativeEventSubscription } from 'react-native';
+
+let backgroundMoveTime: number | null = null;
+
+abstract class TimerBase {
+  protected listener?: NativeEventSubscription;
+  protected timerId?: number;
+  protected onFinish?: Function;
+  protected callback: Function;
+
+  constructor(
+    callback: Function,
+    startImmediately: boolean,
+    onFinish?: Function,
+  ) {
+    this.callback = callback;
+    this.onFinish = onFinish;
```

Comment:
I think this function should not be a part of the abstract class. 
We should implement it at the children class level